nixos/modules/services/mail/maddy.nix

```
@@ -335,12 +335,13 @@ in {
       };
 
       secrets = lib.mkOption {
-        type = lib.types.path;
+        type = with types; listOf path;
```

Thinking about it, that way we break backwards compatibility with setups that use the old style of config declaration (blocking(?) backportability).
That way we can keep compatibility:
```suggestion
        type = with types; (either (nullOr path) (listOf path)) // {
          description = "list of path";
        };
        apply = x: if isList x then x else singleton (warn "Passing a single path to `services.maddy.secrets' is deprecated. Please pass a list of paths instead." x);
```

---

This feels too complicated, or we only support listing a single file as it was?

---

Then I'd prefer going with the list and just let the error message speak for itself.